nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -742,6 +743,40 @@ in
         '';
       };
 
+    virtualisation.directBoot = {
+      enable =
+        mkOption {
+          type = types.bool;
+          default = !cfg.useBootLoader;
+          defaultText = "!cfg.useBootLoader";
+          description =
+            lib.mdDoc ''
+              If enabled, the virtual machine will direct boot the system
+              by passing the kernel, initrd and relevant parameters to QEMU.
+
+              If you want to use a bootloader, use `virtualisation.useBootLoader`,
+              if you want to test netboot, you might be interested into disabling this.
```

Comment:
Use a period `.` to separate sentences.

> If you want to use a bootloader, use `virtualisation.useBootLoader`,

Are you implying I shouldn't set `virtualisation.directBoot` then? I think this raises more questions than it answers. The relation between `directBoot` and `useBootLoader` is already described a bit through the `defaultText`, isn't it? Might that be sufficient?

> if you want to test netboot, you might be interested into disabling this.

"this" is ambiguous.

Question: what if the boot "method" of qemu was specified through an `enum ["net", "kernel", "drive"]` or something? Would that make sense? Would that be easier or harder to implement?
Note that `enum`s can be extended through option merging, so other boot methods could be separated into separate modules that add themselves to the enum by saying `virtualisation.bootMethod = mkOption { type = enum ["net"]; }` for example.
At some point it's better to require the user to be explicit rather than trying to guess based on the options they've set. I've experienced that in arion (although admittedly not implemented this change there yet). Has the time come to make this change here?

Reply:
I agree this is where we should go @ bootMethod, but this comes clear only as we pile up on those ugly abstractions atm.

(also, I believe it should be: "direct", "drive", "firmware" (maybe an extra optionrom? idk) ?)

Reply:
> Are you implying I shouldn't set virtualisation.directBoot then? I think this raises more questions than it answers. The relation between directBoot and useBootLoader is already described a bit through the defaultText, isn't it? Might that be sufficient?

actually, you can set directboot and use a bootloader, you're right. I will rework that :)